recipes/nlohmann_json/all/conanfile.py

```diff
@@ -11,26 +11,33 @@ class NlohmannJsonConan(ConanFile):
     name = "nlohmann_json"
     homepage = "https://github.com/nlohmann/json"
     description = "JSON for Modern C++ parser and generator."
-    topics = ("jsonformoderncpp", "nlohmann_json", "json", "header-only")
+    topics = "json", "header-only"
     url = "https://github.com/conan-io/conan-center-index"
     license = "MIT"
     settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def package_id(self):
         self.info.clear()
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            check_min_cppstd(self, 11)
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._minimum_cpp_standard)
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version],
             destination=self.source_folder, strip_root=True)
 
+    def generate(self):
+        pass
```

Comment:
why empty method is needed? if it does nothing, then it's useless...

Reply:
Try building without the method when using Conan 1.53.0.

```
nlohmann_json/3.11.2: Building your package in /var/home/jordan/.conan/data/nlohmann_json/3.11.2/_/_/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
nlohmann_json/3.11.2: Generator txt created conanbuildinfo.txt
Traceback (most recent call last):
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/command.py", line 2216, in run
    method(args[0][1:])
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/command.py", line 380, in create
    info = self._conan.create(args.path, name=name, version=version, user=user,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/conan_api.py", line 93, in wrapper
    return f(api, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/conan_api.py", line 388, in create
    create(self.app, ref, graph_info, remotes, update, build_modes,
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/cmd/create.py", line 72, in create
    install_build_and_test(app, test_conanfile_path,
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/cmd/test.py", line 35, in install_build_and_test
    install_folder, conanfile = deps_install(app=app,
                                ^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/manager.py", line 94, in deps_install
    installer.install(deps_graph, remotes, build_modes, update, profile_host, profile_build,
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/installer.py", line 317, in install
    self._build(nodes_by_level, keep_build, root_node, profile_host, profile_build,
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/installer.py", line 466, in _build
    self._handle_node_cache(node, keep_build, processed_package_refs, remotes)
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/installer.py", line 559, in _handle_node_cache
    pref = self._build_package(node, output, keep_build, remotes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/installer.py", line 602, in _build_package
    pref = builder.build_package(node, keep_build, self._recorder, remotes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/installer.py", line 227, in build_package
    self._build(conanfile, pref)
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/installer.py", line 138, in _build
    write_toolchain(conanfile, conanfile.generators_folder, self._output)
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/generators/__init__.py", line 257, in write_toolchain
    with chdir(path):
  File "/usr/lib64/python3.11/contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "/var/home/jordan/.local/lib/python3.11/site-packages/conans/client/tools/files.py", line 27, in chdir
    os.chdir(newdir)
FileNotFoundError: [Errno 2] No such file or directory: '/var/home/jordan/.conan/data/nlohmann_json/3.11.2/_/_/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/build-debug/conan'
```

Reply:
sounds like a bug, is it reported? @uilianries

Reply:
It's not reported AFAIK. We only declared those empty method to avoid Conan warnings, but it hided a basic bug 😱

Reply:
@jwillikers Could you please report it to conan client? It deserves some investigation.

Reply:
Done: conan-io/conan#12383